plugin/src/main/groovy/com/novoda/buildproperties/internal/BuildPropertiesException.groovy

```diff
@@ -0,0 +1,63 @@
+package com.novoda.buildproperties.internal
+
+import static java.util.Collections.emptyList
+import static java.util.Collections.singletonList
+
+class BuildPropertiesException extends Exception {
+
+    public static final BuildPropertiesException EMPTY = new BuildPropertiesException(emptyList(), emptyList())
+
+    private final List<Throwable> exceptions
+    private final List<String> additionalMessages
+
+    static BuildPropertiesException from(String message, String additionalMessage = '') {
+        from(new BuildPropertiesException(new Exception(message), additionalMessage))
+    }
+
+    static BuildPropertiesException from(Throwable throwable) {
```

Comment:
should there be parity here with the superclass of BPE - i.e. should this be `Exception` or should BPE extend `Throwable`?